### ISSUE TITLE
GetGuidOfProject called on the wrong thread -> handled exception

### ISSUE THREAD

**Issue by duncanp-sonar** [SonarSource/sonarlint-visualstudio#3195]
### Description

The `TextBufferIssueTracker` calls `GetProjectGuid` on the wrong thread, causing VS to throw.
The exception is caught and logged -> no crash.

Unclear what the user impact is - analysis is not run for the specified file?

### Repro steps
Not sure - was debugging with standalone slvs-samples solution open and happened to notice this in the OutputWindow.

There were multiple file tabs visible, but I'd only opened a couple of files. I think the file `CognitiveComplexity.ts` file was top-most.

#### Stack trace
```
DEBUG: Failed to calculate project guid of file D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\CSharp_SDKStyleProject_ConsoleApp\ts\CognitiveComplexity.ts, error occurred: System.Runtime.InteropServices.COMException (0x8001010E): GetGuidOfProjectFile must be called on the UI thread.
   at Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName)
   at Microsoft.VisualStudio.CommonIDE.Solutions.Solution.Microsoft.VisualStudio.Shell.Interop.IVsSolution5.GetGuidOfProjectFile(String pszProjectFile)
   at SonarLint.VisualStudio.Integration.Vsix.TextBufferIssueTracker.GetProjectGuid()
```

**Comment by duncanp-sonar** [comment]
Another repro / call stack:
* opened a solution with open file -> analysis triggered.

#### Stack trace
```
Analyzing D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp
Refreshing PCH file for D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp. PCH file location: C:\Users\duncan.pocklington\AppData\Local\Temp\SLVS\PCH\876334b6-77f9-4f1c-9ba6-1a3611d08c5f\PCH.preamble
DEBUG: Failed to calculate project guid of file D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp, error occurred: System.Runtime.InteropServices.COMException (0x8001010E): GetGuidOfProjectFile must be called on the UI thread.
   at Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName)
   at Microsoft.VisualStudio.CommonIDE.Solutions.Solution.Microsoft.VisualStudio.Shell.Interop.IVsSolution5.GetGuidOfProjectFile(String pszProjectFile)
   at SonarLint.VisualStudio.Integration.Vsix.TextBufferIssueTracker.GetProjectGuid()
DEBUG: Failed to calculate project guid of file D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp, error occurred: System.Runtime.InteropServices.COMException (0x8001010E): GetGuidOfProjectFile must be called on the UI thread.
   at Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName)
   at Microsoft.VisualStudio.CommonIDE.Solutions.Solution.Microsoft.VisualStudio.Shell.Interop.IVsSolution5.GetGuidOfProjectFile(String pszProjectFile)
   at SonarLint.VisualStudio.Integration.Vsix.TextBufferIssueTracker.GetProjectGuid()
DEBUG: Failed to calculate project guid of file D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp, error occurred: System.Runtime.InteropServices.COMException (0x8001010E): GetGuidOfProjectFile must be called on the UI thread.
   at Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName)
   at Microsoft.VisualStudio.CommonIDE.Solutions.Solution.Microsoft.VisualStudio.Shell.Interop.IVsSolution5.GetGuidOfProjectFile(String pszProjectFile)
   at SonarLint.VisualStudio.Integration.Vsix.TextBufferIssueTracker.GetProjectGuid()
DEBUG: Failed to calculate project guid of file D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp, error occurred: System.Runtime.InteropServices.COMException (0x8001010E): GetGuidOfProjectFile must be called on the UI thread.
   at Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName)
   at Microsoft.VisualStudio.CommonIDE.Solutions.Solution.Microsoft.VisualStudio.Shell.Interop.IVsSolution5.GetGuidOfProjectFile(String pszProjectFile)
   at SonarLint.VisualStudio.Integration.Vsix.TextBufferIssueTracker.GetProjectGuid()
DEBUG: Failed to calculate project guid of file D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp, error occurred: System.Runtime.InteropServices.COMException (0x8001010E): GetGuidOfProjectFile must be called on the UI thread.
   at Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName)
   at Microsoft.VisualStudio.CommonIDE.Solutions.Solution.Microsoft.VisualStudio.Shell.Interop.IVsSolution5.GetGuidOfProjectFile(String pszProjectFile)
   at SonarLint.VisualStudio.Integration.Vsix.TextBufferIssueTracker.GetProjectGuid()
DEBUG: Failed to calculate project guid of file D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp, error occurred: System.Runtime.InteropServices.COMException (0x8001010E): GetGuidOfProjectFile must be called on the UI thread.
   at Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName)
   at Microsoft.VisualStudio.CommonIDE.Solutions.Solution.Microsoft.VisualStudio.Shell.Interop.IVsSolution5.GetGuidOfProjectFile(String pszProjectFile)
   at SonarLint.VisualStudio.Integration.Vsix.TextBufferIssueTracker.GetProjectGuid()
DEBUG: Failed to calculate project guid of file D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp, error occurred: System.Runtime.InteropServices.COMException (0x8001010E): GetGuidOfProjectFile must be called on the UI thread.
   at Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName)
   at Microsoft.VisualStudio.CommonIDE.Solutions.Solution.Microsoft.VisualStudio.Shell.Interop.IVsSolution5.GetGuidOfProjectFile(String pszProjectFile)
   at SonarLint.VisualStudio.Integration.Vsix.TextBufferIssueTracker.GetProjectGuid()
DEBUG: Failed to calculate project guid of file D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp, error occurred: System.Runtime.InteropServices.COMException (0x8001010E): GetGuidOfProjectFile must be called on the UI thread.
   at Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName)
   at Microsoft.VisualStudio.CommonIDE.Solutions.Solution.Microsoft.VisualStudio.Shell.Interop.IVsSolution5.GetGuidOfProjectFile(String pszProjectFile)
   at SonarLint.VisualStudio.Integration.Vsix.TextBufferIssueTracker.GetProjectGuid()
DEBUG: Failed to calculate project guid of file D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp, error occurred: System.Runtime.InteropServices.COMException (0x8001010E): GetGuidOfProjectFile must be called on the UI thread.
   at Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName)
   at Microsoft.VisualStudio.CommonIDE.Solutions.Solution.Microsoft.VisualStudio.Shell.Interop.IVsSolution5.GetGuidOfProjectFile(String pszProjectFile)
   at SonarLint.VisualStudio.Integration.Vsix.TextBufferIssueTracker.GetProjectGuid()
DEBUG: Failed to calculate project guid of file D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp, error occurred: System.Runtime.InteropServices.COMException (0x8001010E): GetGuidOfProjectFile must be called on the UI thread.
   at Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName)
   at Microsoft.VisualStudio.CommonIDE.Solutions.Solution.Microsoft.VisualStudio.Shell.Interop.IVsSolution5.GetGuidOfProjectFile(String pszProjectFile)
   at SonarLint.VisualStudio.Integration.Vsix.TextBufferIssueTracker.GetProjectGuid()
DEBUG: Failed to calculate project guid of file D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp, error occurred: System.Runtime.InteropServices.COMException (0x8001010E): GetGuidOfProjectFile must be called on the UI thread.
   at Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName)
   at Microsoft.VisualStudio.CommonIDE.Solutions.Solution.Microsoft.VisualStudio.Shell.Interop.IVsSolution5.GetGuidOfProjectFile(String pszProjectFile)
   at SonarLint.VisualStudio.Integration.Vsix.TextBufferIssueTracker.GetProjectGuid()
DEBUG: Failed to calculate project guid of file D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp, error occurred: System.Runtime.InteropServices.COMException (0x8001010E): GetGuidOfProjectFile must be called on the UI thread.
   at Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName)
   at Microsoft.VisualStudio.CommonIDE.Solutions.Solution.Microsoft.VisualStudio.Shell.Interop.IVsSolution5.GetGuidOfProjectFile(String pszProjectFile)
   at SonarLint.VisualStudio.Integration.Vsix.TextBufferIssueTracker.GetProjectGuid()
DEBUG: Failed to calculate project guid of file D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp, error occurred: System.Runtime.InteropServices.COMException (0x8001010E): GetGuidOfProjectFile must be called on the UI thread.
   at Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName)
   at Microsoft.VisualStudio.CommonIDE.Solutions.Solution.Microsoft.VisualStudio.Shell.Interop.IVsSolution5.GetGuidOfProjectFile(String pszProjectFile)
   at SonarLint.VisualStudio.Integration.Vsix.TextBufferIssueTracker.GetProjectGuid()
DEBUG: Failed to calculate project guid of file D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp, error occurred: System.Runtime.InteropServices.COMException (0x8001010E): GetGuidOfProjectFile must be called on the UI thread.
   at Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName)
   at Microsoft.VisualStudio.CommonIDE.Solutions.Solution.Microsoft.VisualStudio.Shell.Interop.IVsSolution5.GetGuidOfProjectFile(String pszProjectFile)
   at SonarLint.VisualStudio.Integration.Vsix.TextBufferIssueTracker.GetProjectGuid()
DEBUG: Failed to calculate project guid of file D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp, error occurred: System.Runtime.InteropServices.COMException (0x8001010E): GetGuidOfProjectFile must be called on the UI thread.
   at Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName)
   at Microsoft.VisualStudio.CommonIDE.Solutions.Solution.Microsoft.VisualStudio.Shell.Interop.IVsSolution5.GetGuidOfProjectFile(String pszProjectFile)
   at SonarLint.VisualStudio.Integration.Vsix.TextBufferIssueTracker.GetProjectGuid()
DEBUG: Failed to calculate project guid of file D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp, error occurred: System.Runtime.InteropServices.COMException (0x8001010E): GetGuidOfProjectFile must be called on the UI thread.
   at Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName)
   at Microsoft.VisualStudio.CommonIDE.Solutions.Solution.Microsoft.VisualStudio.Shell.Interop.IVsSolution5.GetGuidOfProjectFile(String pszProjectFile)
   at SonarLint.VisualStudio.Integration.Vsix.TextBufferIssueTracker.GetProjectGuid()
DEBUG: Failed to calculate project guid of file D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp, error occurred: System.Runtime.InteropServices.COMException (0x8001010E): GetGuidOfProjectFile must be called on the UI thread.
   at Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName)
   at Microsoft.VisualStudio.CommonIDE.Solutions.Solution.Microsoft.VisualStudio.Shell.Interop.IVsSolution5.GetGuidOfProjectFile(String pszProjectFile)
   at SonarLint.VisualStudio.Integration.Vsix.TextBufferIssueTracker.GetProjectGuid()
Finished analyzing D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp, analysis time: 30.712s
Found 17 issue(s) for D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp
DEBUG: [VCX:VcxRequestFactory] [Thread id: 73] Trying to create request for D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp
DEBUG: [VCX:VcxRequestFactory] [Thread id: 73] 	Created request successfully
Refreshing PCH file for D:\repos\sq\slvs.samples\standalone\vs2022\NormalSolution\Cpp_WindowsProject_VS2022\Rules\S3135_v6_36.cpp. PCH file location: C:\Users\duncan.pocklington\AppData\Local\Temp\SLVS\PCH\876334b6-77f9-4f1c-9ba6-1a3611d08c5f\PCH.preamble
```
